include/zephyr/bluetooth/audio/has.h

```diff
@@ -72,10 +72,20 @@ struct bt_has_register_param {
 	/** Hearing Aid Type value */
 	enum bt_has_hearing_aid_type type;
 
-	/** Preset Synchronization Support. Only applicable for binaural hearing aids. */
+	/**
+	 * @brief Preset Synchronization Support.
+	 *
+	 * Only applicable if @p type is @ref BT_HAS_HEARING_AID_TYPE_BINAURAL
+	 * and @kconfig{CONFIG_BT_HAS_PRESET_COUNT} is non-zero.
```

Comment:
Should we guard `preset_sync_support` and `independent_presets` with `CONFIG_BT_HAS_PRESET_COUNT > 0`?

Reply:
I have mixed feelings. This would require to guard the code in `has.c` as well. Leaving it non-guarded makes it possible to use `IS_ENABLED` macro that compiles the code even if flag is not set (later the code is removed by linker). 
Those are registration parameters, well documented, so I would not bother guarding those.

Reply:
> Those are registration parameters, well documented, so I would not bother guarding those.

True. The only reason why we would guard it, is to make it 2 bytes smaller. However, given that it is a short-lived variable (e.g. can just be stack allocated), the advantage of guarding them is minimal. 

For me it is OK to keep as is